dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/TracedEncodingTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.dialogue.serde;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import com.google.common.io.ByteStreams;
+import com.palantir.dialogue.TypeMarker;
+import com.palantir.tracing.AlwaysSampler;
+import com.palantir.tracing.Tracer;
+import com.palantir.tracing.api.Span;
+import com.palantir.tracing.api.SpanObserver;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+class TracedEncodingTest {
+
+    @Test
+    void testTypeString_string() {
+        // non-generic types use the simple name
+        assertThat(TracedEncoding.toString(new TypeMarker<String>() {})).isEqualTo("String");
+    }
+
+    @Test
+    void testTypeString_stringList() {
+        // It's more complicated to generate 'List<String>' from Type, that can be done later if necessary
+        assertThat(TracedEncoding.toString(new TypeMarker<List<String>>() {}))
+                .isEqualTo("java.util.List<java.lang.String>");
+    }
+
+    @Test
+    void testSerializerOperationName() {
+        Tracer.setSampler(AlwaysSampler.INSTANCE);
+        Tracer.getAndClearTrace();
+        Encoding.Serializer<String> serializer =
+                new TracedEncoding(new StubEncoding()).serializer(new TypeMarker<String>() {});
+        SpanObserver mockObserver = mock(SpanObserver.class);
+        Tracer.subscribe("test", mockObserver);
```

Comment:
could we hoist all this setup code to an `@Before` so that it's easier to skim the tests and see what the difference is in each of them?

Reply:
Unfortunately it only applies to two of the four tests in this class. We could split them into separate classes, but it becomes more difficult to jump from TracedEncoding to its tests :-/
Fwiw this test is identical to the traced encoding test for conjure-undertow with the "Undertow" substring replaced with "Dialogue" ;-)

Reply:
Personally I'd still hoist them. Things like `Tracer.setSample(AlwaysSample.INSTANCE)` don't hurt your first two tests, they just seem like a sane default for this whole file?